install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,665 @@ toc::[]
-setup to xref:using-stick-tables[use stick-tables] that synchronizes between
-peers. The router plug-in provides all the facilities necessary to make this
-customization.
+The default HAProxy router is intended to satisfy the needs of most users.
+However, it does not expose all of the capability of HAProxy and users
+may need to modify the router for their own needs.
+
+You may need to implement new features within the application back-ends, or
+modify the current operation.  For example, it might
+be desirable in a highly-available setup to use
+xref:using-stick-tables[stick-tables] that synchronizes between peers. The
+router plug-in provides all the facilities necessary to make this customization.
+
+The router pod uses a template file to create the needed HAProxy configuration file.
+The template file is a
+link:http://golang.org/pkg/text/template/[*golang* template].
+When processing the template, the router has access to {product-title}
+information including the router's deployment configuration,
+the set of admitted routes and some helper functions.
+
+When the router pod starts and every time it reloads, it creates a HAProxy
+configuration file and then it starts HAProxy.  The
+link:https://cbonte.github.io/haproxy-dconv/configuration-1.5.html[Haproxy configuration manual]
+describes all of the features of HAProxy and how to construct a valid
+configuration file.
+
+A *ConfigMap* can be used to add the new template to the router pod. With
+this approach, the router deployment configuration is modified to mount the
+xref:using-configmap-replace-template[*configMap*] as a volume in the router
+pod.  The `TEMPLATE_FILE` environment variable is set to the full path name
+of the template file in the router pod.
+
+An alternative approach is to modify the *haproxy-template.config* file
+and xref:rebuilding-your-router[rebuild] the router image. The new image
+is pushed to the the cluster repositiory and the routers deployment
+configuration *image:* field is updated with the new name. When the cluster is
+updated the image needs to be rebuilt and pushed.
+
+In either case the router pod starts with the template file.
 
 [[obtaining-router-configuration-template]]
 == Obtaining the Router Configuration Template
 
-You can obtain a new *_haproxy-config.template_* file from the latest router
-image by running:
+The HAProxy template file is fairly large and complex. For some changes, it may
+be easier to modify the existing template rather than writing a complete
+replacement.  You can obtain a *_haproxy-config.template_* file from a running
+router by logging onto the node that is running the router:
 
 ----
 ifdef::openshift-enterprise[]
 # docker run --rm --interactive=true --tty --entrypoint=cat \
-    registry.access.redhat.com/openshift3/ose-haproxy-router:v3.0.2.0 haproxy-config.template
+    registry.access.redhat.com/openshift3/ose-haproxy-router:v3.5.0.0 haproxy-config.template
```

Comment:
Yeah, its easier to just get it from a current running pod and use the `oc` set. Maybe: 
`oc rsh <router-pod-name>  cat haproxy-config.template > localdir/haproxy-config.template` ?